api/applications/applicationRouter.js

```diff
@@ -320,183 +170,67 @@ router.post(
 
 /**
  * @swagger
- * /application/{approve/:id}:
+ * /application/{update-validate_status/:id}:
  *  put:
- *    summary: Registers profile with okta, updates application ticket "approved" to true, returns a temporary password
- *    description: If a pending application with this profile ID exists, middlware will fetch that profile from the database. This object is passed along into registerOktaUser where its shaped and sent to Oktas API using their clients createUser method.
+ *    summary: Updates validate_status of a given user
+ *    description: user is fetched from MongoDB by the profile_id in req.params, and is modified by the validate_status selected by the user (approved/rejected)
  *    tags:
  *      - application
  *    security:
- *      - okta: [authRequired, adminRequired]
+ *      - auth0: [authRequired, adminRequired]
  *    parameters:
  *      - in: param
  *        name: profile id
- *        schema:
- *          type: string
- *        description: Profile ID of pending applicant
  *    responses:
  *      '200':
- *        description: Response from successful put
+ *        description: Response from successful post to MongoDB
  *        content:
  *          application/json:
  *            schema:
  *              type: object
- *              properties:
- *                tempPassword:
- *                  description: temporary password for admins to give to approved users
- *                  type: string
  *              items:
- *                $ref: '#/components/schemas/Application'
- *              example:
- *                message: "3sdfi324"
+ *                status: 200
+ *                message: {
+ *                  result: true
+ *                    }
  *      '401':
  *        $ref: '#/components/responses/UnauthorizedError'
- */
 
-// registers a user with okta and approves their application ticket
+NOTE: FE discerns if the user is a mentor/mentee and sends back the appropriate shape: 
+mentor = {validate_status & current_company }
+mentee = {validate_status}
+this endpoint can then send the validate status to the appropriate endpoint depending on whether the current_company is present or not
+  
+authRequired, and AdminRequired imports have been left (commented out) because they will likely be needed when auth0 is implemented.
 
-router.put(
-  '/approve/:id',
-  authRequired,
-  adminRequired,
-  checkApplicationExists,
-  findProfile,
-  registerOktaUser,
-  (req, res, next) => {
-    const application_id = req.application.application_id;
+*/
+router.post('/update-validate_status/:profile_id', (req, res, next) => {
+  const isMentor = req.body.current_company;
+  const { profile_id } = req.params;
```

Comment:
It might be a good idea to have a middleware that checks if a profile_id exists in the DB and return a message like 'user not found'. Otherwise these errors will fall in the catch block along with other potential errors (might make it more difficult to parse the errors)

Reply:
so that's an issue that we'll need to correct/solve with @BrokenShell  and the DS team--I think... As the DS API currently stands, I can send an incorrect profile_id to the read/mentor or mentee endpoints and I will get a 200 response that is an empty array.
Though I guess I could work with that as it is--make the call in the middleware and if the result is an empty array, then I send it back as an error 'user not found'. That could work, I think.